src/acp/policy.test.ts

```diff
@@ -209,6 +226,774 @@ describe("removePolicy", () => {
+      mockedResourceWithAcr,
+      `${getSourceUrl(mockedAcr)}#policy`
+    );
+
+    const updatedPolicy = removeUrl(
+      mockedPolicy,
+      somePredicate,
+      "https://example.test"
+    );
+
+    const updatedResourceWithAcr = setResourcePolicy(
+      mockedResourceWithAcr,
+      updatedPolicy
+    );
+
+    const policyAfterUpdate = getResourcePolicy(
+      updatedResourceWithAcr,
+      "policy"
+    );
+    expect(getUrl(policyAfterUpdate!, somePredicate)).toBeNull();
+  });
+
+  it("applies the Policy to the Resource", () => {
+    const mockedAcr = mockAcrFor("https://some.pod/resource");
+    const mockedResourceWithAcr = addMockAcrTo(
+      mockSolidDatasetFrom("https://some.pod/resource"),
+      mockedAcr
+    );
+    const mockedPolicy = createResourcePolicyFor(
+      mockedResourceWithAcr,
+      "policy"
+    );
+
+    const updatedResourceWithAcr = setResourcePolicy(
+      mockedResourceWithAcr,
+      mockedPolicy
+    );
+
+    expect(getPolicyUrlAll(updatedResourceWithAcr)).toEqual([
+      `${getSourceUrl(mockedAcr)}#policy`,
+    ]);
+  });
+});
+
+describe("setResourceAcrPolicy", () => {
+  it("replaces existing instances of the set Access Policy", () => {
+    const somePredicate = "https://some.vocab/predicate";
+    let mockedAcr = mockAcrFor("https://some.pod/resource");
+    let mockedPolicy = createThing({
+      url: `${getSourceUrl(mockedAcr)}#policy`,
+    });
+    mockedPolicy = setUrl(mockedPolicy, rdf.type, acp.Policy);
+    mockedPolicy = setUrl(mockedPolicy, somePredicate, "https://example.test");
+    mockedAcr = setThing(mockedAcr, mockedPolicy);
+    let mockedResourceWithAcr = addMockAcrTo(
+      mockSolidDatasetFrom("https://some.pod/resource"),
+      mockedAcr
+    );
+    mockedResourceWithAcr = addAcrPolicyUrl(
+      mockedResourceWithAcr,
+      `${getSourceUrl(mockedAcr)}#policy`
+    );
+
+    const updatedPolicy = removeUrl(
+      mockedPolicy,
+      somePredicate,
+      "https://example.test"
+    );
+
+    const updatedResourceWithAcr = setResourceAcrPolicy(
+      mockedResourceWithAcr,
+      updatedPolicy
+    );
+
+    const policyAfterUpdate = getResourceAcrPolicy(
+      updatedResourceWithAcr,
+      "policy"
+    );
+    expect(getUrl(policyAfterUpdate!, somePredicate)).toBeNull();
+  });
+
+  it("applies the Policy to the Resource's ACR", () => {
+    const mockedAcr = mockAcrFor("https://some.pod/resource");
+    const mockedResourceWithAcr = addMockAcrTo(
+      mockSolidDatasetFrom("https://some.pod/resource"),
+      mockedAcr
+    );
+    const mockedPolicy = createResourcePolicyFor(
+      mockedResourceWithAcr,
+      "policy"
+    );
+
+    const updatedResourceWithAcr = setResourceAcrPolicy(
+      mockedResourceWithAcr,
+      mockedPolicy
+    );
+
+    expect(getAcrPolicyUrlAll(updatedResourceWithAcr)).toEqual([
+      `${getSourceUrl(mockedAcr)}#policy`,
+    ]);
+  });
+});
+
+describe("removeResourcePolicy", () => {
```

Comment:
Technically, this doesn't check if it does not remove a non-policy with a matching name, but that's a bit nitpicky.

Reply:
On the other hand, it actually did remove a non-Policy with a matching name, which I don't think it should be doing. Could be important when people e.g. mix up the names of their Rules and Policies. b2c884e21cb0ddd23f9ac6b39565e5a6436dc581 and 354fd10ecd7800588859d6817f024e4a8150d748.